ports/nrf/drivers/usb/usb_cdc.c

```diff
@@ -0,0 +1,228 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2019 Ha Thach (tinyusb.org)
+ * Copyright (c) 2019 Glenn Ruben Bakke
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * This file is part of the TinyUSB stack.
+ */
+
+#include "py/mphal.h"
+
+#if USB_CDC
+
+#include "tusb.h"
+#include "nrfx.h"
+#include "nrfx_power.h"
+#include "nrfx_uart.h"
+#include "py/ringbuf.h"
+
+#ifdef BLUETOOTH_SD
+#include "nrf_sdm.h"
+#include "nrf_soc.h"
+#include "ble_drv.h"
+#endif
+
+extern void tusb_hal_nrf_power_event(uint32_t event);
+
+static void cdc_task(void);
+
+uint8_t rx_ringbuf_array[4096];
+uint8_t tx_ringbuf_array[4096];
```

Comment:
I've put quite large buffers to handle good REPL paste feeling. This might be a bit too much.

Reply:
stm32 uses 1024 bytes for each, so probably that would be plenty.

Reply:
Amended change to use 1024, ringbuffer size. Also made this group of variables static. Removed unused rx_buf and tx_buf.